libs/libpcache/src/config/config.cpp

```diff
@@ -139,6 +198,105 @@ void Config::load_config (string filename) {
             this->LOG_FILE = logfile;
         }
     }
+
+    unordered_map<string,vector<Fault*>> faults; //(path,[Fault1,Fault2,...])
+    if (data.contains ("injection")) {
+        const auto& programmed_injections = toml::find<toml::array>(data,"injection");
+	
+        for (const auto& injection : programmed_injections) {
```

Comment:
I think this part of the code looks very confusing due to the repeated if statements, can you add inline documentation explaining each step?

Reply:
I have incorporated only some general information in the comments, because the majority of the "if" statements are used for raising errors, and each accompanying message elucidates the specific purpose behind these "if" conditions.